src/web_or_worker_shell_read.js

```diff
@@ -22,17 +22,12 @@
   }
 
   readAsync = (url, onload, onerror) => {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', url, true);
-    xhr.responseType = 'arraybuffer';
-    xhr.onload = () => {
-      if (xhr.status == 200 || (xhr.status == 0 && xhr.response)) { // file URLs can return 0
-        onload(xhr.response);
-        return;
+    fetch(url)
```

Comment:
Can you also update the above two uses of `XMLHttpRequest` ?

Reply:
Looks like the two earlier calls are synchronous, it might take some more re-engineering to get those converted.